openfl/component/aggregator/straggler_handling.py

```diff
@@ -140,3 +192,71 @@ def __minimum_collaborators_reported(self, num_collaborators_done) -> bool:
                 False otherwise.
         """
         return num_collaborators_done >= self.minimum_reporting
+
+
+class PercentagePolicy(StragglerHandlingPolicy):
+    """Percentage based Straggler Handling function."""
+
+    def __init__(self, percent_collaborators_needed=1.0, minimum_reporting=1, **kwargs):
+        """Initialize a PercentagePolicy object.
+
+        Args:
+            percent_collaborators_needed (float, optional): The percentage of
+                collaborators needed. Defaults to 1.0.
+            minimum_reporting (int, optional): The minimum number of
+                collaborators that should report. Defaults to 1.
+            **kwargs: Variable length argument list.
+        """
+        if minimum_reporting <= 0:
+            raise ValueError("minimum_reporting must be >0")
```

Comment:
One liner: `assert minimum_reporting > 0, "Minimum reporting must be >0"`

Reply:
IMO, assert statements are primarily intended for debugging purposes. 
Using raise with a ValueError more clearly indicates that the error is due to an invalid argument